aten/src/ATen/hip/impl/HIPGuardImplMasqueradingAsCUDA.h

```diff
@@ -213,12 +213,36 @@ struct HIPGuardImplMasqueradingAsCUDA final : public c10::impl::DeviceGuardImplI
     hip_stream.synchronize();
   }
 
+  void synchronizeEvent(void* event) const override {
+    if (!event)
+      return;
+    hipEvent_t hip_event = static_cast<hipEvent_t>(event);
+    C10_HIP_CHECK(hipEventSynchronize(hip_event));
+  }
+
   void recordDataPtrOnStream(
     const c10::DataPtr& data_ptr,
     const Stream& stream) const override {
     HIPStreamMasqueradingAsCUDA hip_stream{stream};
     HIPCachingAllocatorMasqueradingAsCUDA::recordStreamMasqueradingAsCUDA(data_ptr, hip_stream);
   }
+
+  double elapsedTime(void* event1, void* event2, const DeviceIndex device_index)
```

Comment:
Can we please make sure the relevant submodule owner has a chance to review the change before we merge it in other submodules?

cc @jeffdaily does this timing code looks good?

Reply:
Nothing jumps out as wrong.  I'll lean on CI to tell the truth :-).

Reply:
> Can we please make sure the relevant submodule owner has a chance to review the change before we merge it in other submodules?
> 
> cc @jeffdaily does this timing code looks good?

Sorry about that...

Reply:
I refer to `destroyEvent` code to guard it not to crate a new cuda context here. https://github.com/pytorch/pytorch/blob/b24ad7eab55eaf660893dddae949fc714e434338/c10/cuda/impl/CUDAGuardImpl.h#L106